veggieseasons/lib/widgets/trivia.dart

```diff
@@ -203,12 +211,34 @@ class _TriviaViewState extends State<TriviaView> {
           CupertinoButton(
             child: Text('Next Question'),
             onPressed: () => setState(() {
-              triviaIndex++;
-              status = PlayerStatus.readyToAnswer;
+              triviaIndex.value++;
+              status.value = PlayerStatus.readyToAnswer;
             }),
           ),
         ],
       ),
     );
   }
+
+  @override
+  RestorationId get restorationId => widget.restorationId;
+}
+
+class RestorablePlayerStatus extends RestorableValue<PlayerStatus> {
+  RestorablePlayerStatus(PlayerStatus value) : super(value);
+
+  @override
+  PlayerStatus deserialize(Object serialized) {
+    return PlayerStatus.values[serialized as int];
+  }
+
+  @override
+  Object serialize() {
+    return value.index;
+  }
+
+  @override
+  void didUpdateValue(PlayerStatus oldValue) {
+    notifyListeners();
+  }
```

Comment:
Can you make a RestorableEnum<T> ?

Reply:
Originally I looked into that, but since enums don't have a common base class other than Object I don't see a good way to do that.